### ISSUE TITLE
4 of 10 cross-validation folds fail to converge at M=150 / 400 draws

### ISSUE THREAD

**Issue by bschilder** [bschilder/genomeOS#111]
The first completed held-out validation run (#109) scored **6 of 10 folds**. The other four were
refused by the §12 convergence gate:

| strategy | scored | failed | failures |
|---|---|---|---|
| spatial | 4/5 | 1 | ESS 85 |
| random | **2/5** | **3** | r_hat 1.065 / ESS 61; ESS 143; ESS 72 |

Config: `n_inducing=150`, `draws=400`, `tune=400`, `chains=4`, beta-binomial, numpyro NUTS on an
A100. Gate: `max_rhat=1.05`, `min_ess=200`.

The `FoldFailure` tolerance (#108) did its job — the run completed and reported the exclusions
rather than dying or silently averaging them in. But **a 40% failure rate is itself a result**,
and it undermines the numbers that did come back in two ways:

1. **The surviving folds are a biased sample.** Folds that converge are plausibly the ones whose
   training set was easier, so scores computed over them flatter the model. A skill of +25.9
   measured on 4 of 5 folds is not the same claim as +25.9 measured on 5 of 5.
2. **Random folds fared worse than spatial ones** (3 failures vs 1), which is the opposite of the
   intuitive ordering — random folds have denser, better-spread training data. Worth
   understanding rather than assuming it is noise.

Every failure is an **ESS** failure; only one also tripped r_hat. That points at sampling
efficiency rather than at multimodality or a broken posterior — the chains are mixing, just not
fast enough to bank 200 effective draws out of 1,600.

## Things to try, cheapest first

- **More draws.** 400 is low. The full-data fits used 400 and passed, but folds train on 80% of
  the data and mix worse. This is the obvious first move and costs only time.
- **Raise `target_accept`.** One divergence was observed on a small test fit; a higher
  `target_accept` trades speed for smaller steps.
- **Check which parameter is starving.** `_check_convergence` reports the worst ESS across all
  parameters but not *which* one. If it is always `concentration` or `cohort_sd`, that is a
  reparameterisation problem, not a budget problem — and cheap to fix. Reporting the offending
  parameter name in `ConvergenceError` would make this self-diagnosing.
- **Reconsider `min_ess=200` per fold.** The threshold was set for a publishable surface. A CV
  fold is not published, so a separate, lower bar may be defensible — but it should be a stated
  decision, not a quiet relaxation.

## Not a reason to weaken the gate

§12 says a non-converged fit is excluded rather than published, and that stays. The fix is to
make the folds converge, not to lower the bar until they pass.

Related: #109, #110, #108.

**Comment by bschilder** [comment]
**This is no longer a cross-validation problem. Full-data fits now fail too.**

First real multi-variant batch run (`scripts/build_surfaces.py`, A100, `n_inducing=150`, `draws=400`, `tune=400`, `chains=4`):

```
0/2 variants fitted (100% excluded)
  excluded 2  did_not_converge

chr11-5227002-T-A          ESS 199 < 200                       1071 observations
phenotype:g6pd-deficiency  r_hat 1.492 > 1.05, ESS 8            910 observations
```

Two different failures wearing the same label.

## HbS: a budget problem, and partly self-inflicted

**ESS 199 against a threshold of 200.** With 400 draws x 4 chains the gate demands 12.5% sampling efficiency, which is tight for a GP with a range, an amplitude, a cohort sd and an overdispersion parameter.

It is also a direct consequence of recovering data: HbS went from 857 to 1,071 surveys and the denominator changed from people-approached to people-typed, so the same configuration that converged before now does not. That is not an argument against the recovery — the frequencies are more correct — but it does mean **the sampling budget was tuned to a smaller, easier dataset** and did not travel.

## G6PD: not a near miss

**r_hat 1.492 with ESS 8** is chains that never found a common distribution. More draws may fix it; they may not. Things worth ruling out in order:

- **Budget** — the cheap test, running now at `draws=1200`.
- **Radius spread.** After recovering administrative centroids the radii span 1 km to 564 km within one variant. §7 places each observation as a disc, so the likelihood now mixes near-point surveys with country-scale ones; that is honest about the data but may be hard to sample.
- **The composite phenotype.** ~200 alleles pooled into one frequency (#116) may simply not behave like a single spatial field, in which case no sampler budget rescues it.
- **Which parameter is starving.** `ConvergenceError` reports the worst r_hat and ESS across all parameters but never says *which*. If it is always `concentration` or `cohort_sd` that is a reparameterisation fix, not a budget one — and cheap. **This should be reported in the error message.**

## The gate itself

`min_ess=200` was chosen for a publishable surface and has never been re-examined. A fit at ESS 199 is not meaningfully worse than one at 201, so the threshold is doing binary work on a continuous quantity. Worth deciding deliberately — but *after* the sampler is fixed, not as a way to make failures disappear. §12 stays: a fit that has not mixed is excluded, not published.

## What worked

The batch and its exclusion list behaved exactly as #40 intended: both variants accounted for, both with a stated reason, the list published as an artifact, and the pod stopped by its watchdog at 1,122s. The failure is legible instead of being an empty output directory.
